var/spack/repos/builtin/packages/perl/package.py

```diff
@@ -69,6 +69,9 @@ class Perl(Package):  # Perl doesn't use Autotools, it should subclass Package
     variant('cpanm', default=True,
             description='Optionally install cpanm with the core packages.')
 
+    variant('useshrplib', default=False,
```

Comment:
Most other packages that have a `+shared` variant use it to specify `--enable-shared`. Regardless, both variable names do the same thing, they build shared libraries. I agree that you could argue either way. In this case, I think `+shared` is a lot more descriptive than `+usershrplib`.